src/components/TagModal.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+	<Modal size="large" @close="onClose">
+		<div class="modal-content">
+			<h2 class="tag-title">
+				{{ t('mail', 'Add tags') }}
+			</h2>
+			<div>
+				<div>
+					<div class="tag-group__bg button"
+						style="background-color: #31CC7C" />
+					<span class="tag-group__label"
+						style="color: #31CC7C">
+						{{
+							t('mail', 'Work')
+						}} </span>
+
+					<button class="add-tag"
+						@click="addTag('$label2')">
+						{{ t('mail','Add') }}
+					</button>
+					<button class="remove-button"
+						@click.prevent.stop="removeTag">
```

Comment:
```suggestion
						@click.prevent.stop="removeTag('$label2')">
```

the argument is missing